automations/cargo.py

```diff
@@ -1,63 +1,52 @@
 from magicbot import StateMachine, state
 
-from components.cargo import Arm, Intake
+from components.cargo import CargoManipulator, Height
 
 
 class CargoManager(StateMachine):
 
-    arm: Arm
-    intake: Intake
-
-    def __init__(self):
-        super().__init__()
-        self.override = False
+    cargo_component: CargoManipulator
 
     def intake_floor(self, force=False):
         self.engage(initial_state="move_to_floor", force=force)
 
     @state(first=True, must_finish=True)
-    def move_to_floor(self):
-        self.next_state("intaking_cargo")
+    def move_to_floor(self, initial_call, state_tm):
+        self.cargo_component.move_to(Height.FLOOR)
+        if self.cargo_component.at_height(Height.FLOOR):
+            self.next_state("intaking_cargo")
 
-    def intake_depot(self, force=False):
-        self.engage(initial_state="move_to_depot", force=force)
+    def outake_cargo_ship(self, force=False):
+        self.engage(initial_state="move_to_cargo_ship", force=force)
 
-    @state
-    def move_to_depot(self):
-        self.next_state("intaking_cargo")
+    @state(must_finish=True)
+    def move_to_cargo_ship(self, initial_call, state_tm):
+        self.cargo_component.move_to(Height.CARGO_SHIP)
+        if self.cargo_component.at_height(Height.CARGO_SHIP):
+            self.next_state("outtaking_cargo")
```

Comment:
Uhm, wouldn't we want to be able to raise the cargo mechanism before we get to the cargo ship?